### ISSUE TITLE
Using `setData` with array does not trigger watchers or update the view

### ISSUE THREAD

**Issue by tbuteler** [vuejs/vue-test-utils#605]
### Version
1.0.0-beta.16

### Reproduction link
[https://github.com/tbuteler/vue-test-utils-issue](https://github.com/tbuteler/vue-test-utils-issue)

### Steps to reproduce
Clone the repository, then run:

```bash
yarn install
yarn run tests
```

### What is expected?
Component:

```Vue
<template>
  <div id="app">
    <ul v-if="items.length > 0">
      <li v-for="item in items">
        {{ item }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  name: 'app',
  data () {
    return {
      loaded: false,
      items: []
    }
  },
  watch: {
    items () {
      this.loaded = true
    }
  }
}
</script>
```

Test:
```javascript
import Vue from 'vue'
import { mount } from '@vue/test-utils'
import App from '../src/App.vue'

describe('App.vue', () => {
  it('array data triggers watchers', () => {
    const wrapper = mount(App)
    wrapper.setData({ items: ['Foo', 'Bar'] })
    expect(wrapper.vm.loaded).toBe(true)
    expect(wrapper.findAll('li').at(0).text()).toBe('Foo')
    expect(wrapper.findAll('li').at(1).text()).toBe('Bar')
  })
})
```
After doing `setData` I expected the HTML to be updated and the watcher (which sets `loaded` to `true`) to be run.

### What is actually happening?
Although the `wrapper.vm.items` property yields what I just set, no updates are triggered on the wrapper and the tests fail.

---
Test passes in 1.0.0-beta.15.

**Comment by eddyerburgh** [comment]
Could be related to https://github.com/vuejs/vue-test-utils/issues/563#issuecomment-388886538

**Comment by eddyerburgh** [comment]
This was fixed in #604 . Although there is another bug with setData, if an array is nested
